src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Attributes/JsonPolymorphicAttribute.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System.Text.Json.Serialization
+{
+    /// <summary>
+    /// When placed on a type, indicates that the type should be serialized polymorphically.
+    /// </summary>
+    [System.Runtime.Versioning.RequiresPreviewFeatures("Polymorphic serialization is currently in preview.")]
+    [AttributeUsage(AttributeTargets.Class | AttributeTargets.Interface, AllowMultiple = false, Inherited = false)]
+    public sealed class JsonPolymorphicAttribute : JsonAttribute
+    {
+        /// <summary>
+        /// Gets or sets a custom type discriminator property name for the polymorhic type.
+        /// Uses the default '$type' property name if left unset.
+        /// </summary>
+        public string? CustomTypeDiscriminatorPropertyName { get; set; }
+
+        /// <summary>
+        /// Gets or sets the behavior when serializing an undeclared derived runtime type.
+        /// </summary>
+        public JsonUnknownDerivedTypeHandling UnknownDerivedTypeHandling { get; set; }
+
+        /// <summary>
+        /// When set to <see langword="true"/>, instructs the deserializer to ignore any
+        /// unrecognized type discriminator id's and reverts to the contract of the base type.
+        /// Otherwise, it will fail the deserialization.
+        /// </summary>
+        public bool IgnoreUnrecognizedTypeDiscriminators { get; set; }
```

Comment:
Why cannot this be another enum position? i.e. `JsonUnknownDerivedTypeHandling.Ignore`

Reply:
The enum concerns handling of unrecognized `object` values, `IgnoreUnrecognizedTypeDiscriminators` concerns handling of unrecognized string identifiers in deserialized JSON. I think they should be configured independently.